HW0/P5/P5.txt

```diff
@@ -1 +1,21 @@
 # Your answers here.
+
+#1. It would take the cashier 255 seconds, because it takes 255 times to sum up all the 256 numbers if add two numbers at a time, and it takes one second to add two numbers.
+
+#2. They can verify the total sum faster than the original cashier. They can do this in 34 seconds. The fastest way is for each cashier to verify the same number of bags: 256/8 = 32, it takes all of them 31 seconds. Then it takes 3 seconds to add the 8 numbers. In total, it takes 34 seconds.
+
+#3. If the store has an infinite number of employees, it takes 8 seconds to verify the sum. First, all 256 bags are verified using 1 second. Then add two numbers at a time, there are 128 counts using another 1 second. Continue this process, it takes 8 steps to add all numbers, which uses 8 seconds.
+
+#4. Suppose the time to count total is x, for these N bags, we have 2^(x) = N if adding two bags at a time. Thus the time to count is log2(N). 
+
+#5. If we also consider the time to communicate, the answer would be the time in #3 plus communication time, which is 8+7 = 15 seconds, since while adding two numbers, it also takes 1 second to communicate.
+
+#6. If the each cashier only takes 1 bag, it will take 256+5 = 261 seconds, since it takes 1 second to hand bags to each cashier, then plus the counting time. When handing bag to cashiers, the cashiers who already got the bags can start counting, thus it save some time. 
+
+If each cashier takes 2 bags, we will need 128 cashiers. The time they need is 128+1+5 = 134 seconds, in which 128 is the handing bag time, 6 is the time for cashiers to count. While they handing bags, the cashiers who already got the bags can start counting, thus it save some time. 
```

Comment:
For this one and the one above, I was thinking the process of handing bags from me to cashiers and the process for cashiers to count could take place at the same time. 
For example, 
1st sec: hand bag to cashier A
2nd sec: hand bag to cashier B + A add two bags
3rd sec: hand bag to cashier C + B add two bags
4th sec: hand bag to cashier D + C add two bags + A and B communicate
... Thus it only takes an extra of 6 seconds besides 128. Welcome for further discussions =)